packages/react-combobox/src/components/ComboButton/useComboButton.ts

```diff
@@ -11,18 +14,50 @@ import type { ComboButtonProps, ComboButtonState } from './ComboButton.types';
  * @param props - props from this instance of ComboButton
  * @param ref - reference to root HTMLElement of ComboButton
  */
-export const useComboButton_unstable = (props: ComboButtonProps, ref: React.Ref<HTMLElement>): ComboButtonState => {
+export const useComboButton_unstable = (
+  props: ComboButtonProps,
+  ref: React.Ref<HTMLButtonElement>,
+): ComboButtonState => {
+  const { placeholder, value } = props;
+
+  const { activeOption, open } = useContextSelector(ComboboxContext, ctx => ({
+    activeOption: ctx.activeOption,
+    open: ctx.open,
+  }));
+
+  const nativeProps = getPartitionedNativeProps({
+    props,
+    primarySlotTagName: 'button',
+  });
+
   return {
-    // TODO add appropriate props/defaults
     components: {
-      // TODO add each slot's element type or component
       root: 'div',
+      content: 'button',
+      dropdownIcon: 'span',
     },
-    // TODO add appropriate slots, for example:
-    // mySlot: resolveShorthand(props.mySlot),
-    root: getNativeElementProps('div', {
-      ref,
-      ...props,
+    root: resolveShorthand(props.root, {
+      required: true,
+      defaultProps: nativeProps.root,
+    }),
+    content: resolveShorthand(props.content, {
+      required: true,
+      defaultProps: {
+        ref,
+        role: 'combobox',
+        type: 'button',
+        'aria-activedescendant': activeOption?.id,
+        'aria-expanded': open,
+        children: value ? value : placeholder,
+        ...nativeProps.primary,
+      },
+    }),
+    dropdownIcon: resolveShorthand(props.dropdownIcon, {
+      required: true,
+      defaultProps: {
+        children: '>',
+      },
     }),
+    open: false,
```

Comment:
```suggestion
    open,
```

Should this use `open` value? Or should it be removed?